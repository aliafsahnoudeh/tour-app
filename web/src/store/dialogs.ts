import { defineStore } from "pinia";
import { ref } from "vue";

export const useDialogsStore = defineStore("dialogs", () => {
  const driverDialog = ref<boolean>(false);
  const deleteItemDialog = ref<boolean>(false);
  const deleteItemsDialog = ref<boolean>(false);

  const setDeleteItemsDialog = (input: boolean) => {
    deleteItemsDialog.value = input;
  };

  const setDeleteItemDialog = (input: boolean) => {
    deleteItemDialog.value = input;
  };

  const setDriverDialog = (input: boolean) => {
    driverDialog.value = input;
  };

  return {
    driverDialog,
    deleteItemDialog,
    deleteItemsDialog,
    setDeleteItemsDialog,
    setDriverDialog,
    setDeleteItemDialog,
  };
});
