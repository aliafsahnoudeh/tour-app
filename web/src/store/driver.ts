import { defineStore } from "pinia";
import { ref } from "vue";
import type DriverModel from "../types/DriverModel";

export const useDriverStore = defineStore("driver", () => {
  const driver = ref<DriverModel | undefined>(undefined);

  const setDriver = (payload: DriverModel | undefined) => {
    driver.value = payload;
  };

  return { driver, setDriver };
});
