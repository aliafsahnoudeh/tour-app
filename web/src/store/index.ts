import { defineStore } from 'pinia'


const useStore = defineStore('store', {
  state: () => {
    return { }
  },
  actions: {
  }
});

export default useStore;
