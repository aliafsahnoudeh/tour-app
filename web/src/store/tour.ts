import { defineStore } from "pinia";
import { ref } from "vue";
import type TourModel from "@/types/TourModel";

export const useTourStore = defineStore("tour", () => {
  const tour = ref<TourModel | undefined>(undefined);

  const setTour = (payload: TourModel | undefined) => {
    tour.value = payload;
  };

  return { tour, setTour };
});
