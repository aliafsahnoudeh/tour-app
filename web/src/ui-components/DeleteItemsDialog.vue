<template>
  <Dialog
    v-model:visible="dialogStore.deleteItemsDialog"
    :style="{ width: '450px' }"
    header="Confirm"
    :modal="true"
    ref="delete_items_dialog"
  >
    <div class="confirmation-content">
      <i class="pi pi-exclamation-triangle mr-3" style="font-size: 2rem" />
      <span>Are you sure you want to delete the selected drivers?</span>
    </div>
    <template #footer>
      <Button
        label="No"
        icon="pi pi-times"
        class="p-button-text"
        @click="dialogStore.deleteItemsDialog = false"
      />
      <Button
        label="Yes"
        icon="pi pi-check"
        class="p-button-text"
        @click="emit('delete')"
      />
    </template>
  </Dialog>
</template>

<script setup lang="ts">
import { defineEmits } from "vue";
import { useDialogsStore } from "../store/dialogs";

const emit = defineEmits(["delete"]);

const dialogStore = useDialogsStore();
</script>
