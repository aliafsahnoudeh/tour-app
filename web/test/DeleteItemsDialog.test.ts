import { describe, expect, test, vi, afterEach, beforeEach } from "vitest";
import { createPinia, setActivePinia } from "pinia";
import { shallowMount } from "@vue/test-utils";
import { nextTick } from "vue";

import PrimeVue from "primevue/config";
import Button from "primevue/button";
import Dialog from "primevue/dialog";

import DeleteItemsDialog from "../src/ui-components/DeleteItemsDialog.vue";
import { useDialogsStore } from "../src/store/dialogs";

describe("DeleteItemsDialog.test", () => {
  let wrapper;

  beforeEach(() => {
    setActivePinia(createPinia());

    wrapper = shallowMount(DeleteItemsDialog, {
      global: {
        components: { Dialog, Button },
        plugins: [PrimeVue],
        mocks: {
          ["$primevue"]: {
            config: {
              ripple: true,
            },
          },
        },
      },
    });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test("match the default snapshot", async () => {
    expect(DeleteItemsDialog).toBeTruthy();

    expect(wrapper.element).toMatchSnapshot();
  });

  test("show the dialog when DeleteItemsDialog is true", async () => {
    expect(DeleteItemsDialog).toBeTruthy();
    const store = useDialogsStore();

    store.setDeleteItemsDialog(true);

    await nextTick();

    expect(wrapper.element).toMatchSnapshot();
  });
});
