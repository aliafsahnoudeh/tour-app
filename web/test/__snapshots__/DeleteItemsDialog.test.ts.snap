// Vitest Snapshot v1

exports[`DeleteItemsDialog.test > match the default snapshot 1`] = `
<dialog-stub
  appendto="body"
  autozindex="true"
  basezindex="0"
  closable="true"
  closeicon="pi pi-times"
  closeonescape="true"
  dismissablemask="false"
  draggable="true"
  header="Confirm"
  keepinviewport="true"
  maximizable="false"
  maximizeicon="pi pi-window-maximize"
  minimizeicon="pi pi-window-minimize"
  minx="0"
  miny="0"
  modal="true"
  position="center"
  showheader="true"
  style="width: 450px;"
  visible="false"
/>
`;

exports[`DeleteItemsDialog.test > show the dialog when DeleteItemsDialog is true 1`] = `
<dialog-stub
  appendto="body"
  autozindex="true"
  basezindex="0"
  closable="true"
  closeicon="pi pi-times"
  closeonescape="true"
  dismissablemask="false"
  draggable="true"
  header="Confirm"
  keepinviewport="true"
  maximizable="false"
  maximizeicon="pi pi-window-maximize"
  minimizeicon="pi pi-window-minimize"
  minx="0"
  miny="0"
  modal="true"
  position="center"
  showheader="true"
  style="width: 450px;"
  visible="true"
/>
`;
